src/libs/MerkleTree.ts

```diff
@@ -17,11 +17,9 @@ export default class MerkleTree {
       throw new Error('Depth must be 20 or less')
     }
 
-    const l = leaves.concat(Array.from(Array(Math.pow(2, depth) - leaves.length), () => utils.zeros(32)))
-
-    this.leaves = l
-    this.layers = [l]
-    this.createHashes(leaves)
+    this.leaves = leaves.concat(Array.from(Array(Math.pow(2, depth) - leaves.length), () => utils.zeros(32)))
+    this.layers = [this.leaves]
+    this.createHashes(this.leaves)
```

Comment:
Didn't verify full logic for MerkleTree creation. Assuming we need to pass the concatenated leaves array while calling `createHashes`.